pkg/cmd/tracerunner.go

```diff
@@ -116,13 +164,54 @@ func (o *TraceRunnerOptions) Run() error {
 		}
 	}()
 
-	c := exec.CommandContext(ctx, o.bpftraceBinaryPath, programPath)
+	var c *exec.Cmd
+	if args == nil || len(*args) == 0 {
+		c = exec.CommandContext(ctx, *binary)
+	} else {
+		c = exec.CommandContext(ctx, *binary, *args)
+	}
+
 	c.Stdout = os.Stdout
 	c.Stdin = os.Stdin
 	c.Stderr = os.Stderr
 	return c.Run()
 }
 
+func (o *TraceRunnerOptions) prepBpfTraceCommand() (*string, *string, error) {
+	programPath := o.program
+
+	// Render $container_pid to actual process pid if scoped to container.
+	podUID, _ := o.parsedSelector.PodUID()
+	container, ok := o.parsedSelector.Container()
+	if ok {
+		pid, err := findPidByPodContainer(podUID, container)
+		if err != nil {
+			return nil, nil, err
+		}
+		if pid == nil {
+			return nil, nil, fmt.Errorf("pid not found")
+		}
+		if len(*pid) == 0 {
+			return nil, nil, fmt.Errorf("invalid pid found")
+		}
+		f, err := ioutil.ReadFile(programPath)
+		if err != nil {
+			return nil, nil, err
+		}
+		programPath = path.Join(os.TempDir(), "program-container.bt")
+		r := strings.Replace(string(f), "$container_pid", *pid, -1)
```

Comment:
This `$container_pid` interpolation is also helpful in BCC programs, could we factor out the string substitution for that into a generic function we can easily re-use in the BCC PR?

Reply:
Will be working on it and will update my example BCC PR as part of it.